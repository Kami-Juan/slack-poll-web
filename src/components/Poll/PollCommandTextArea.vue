<template>
  <div id="poll-command-text-area">
    <b-row align-h="center">
      <b-col cols="12" md="6">
        <span id="instructions">Copy-pasta your command into slack :^)</span>
        <textarea
          :value="value"
          @input="$emit('input', $event.target.value)"
          id="command-input"
        >
        </textarea>
      </b-col>
    </b-row>
  </div>
</template>

<script>
export default {
  name: 'PollCommandTextArea',
  props: ['value']
}
</script>

<style>
#command-input {
  width: 100%;
  height: 92px;
  outline: 0;
  border: 1px solid var(--secondary-border-color);
  border-radius: 4px;
  resize: none;
  padding: 10px 15px 15px 15px;
}

#poll-command-text-area {
  text-align: left;
}
</style>
