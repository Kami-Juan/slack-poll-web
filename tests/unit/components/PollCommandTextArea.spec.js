import BootstrapVue from 'bootstrap-vue'
import { LayoutPlugin, ModalPlugin } from 'bootstrap-vue'

import { mount, createLocalVue } from '@vue/test-utils'
import PollCommandTextArea from './../../../src/components/Poll/PollCommandTextArea'

describe('test PollCommandTextArea.vue', () => {
  const localVue = createLocalVue()

  localVue.use(BootstrapVue)
  localVue.use(LayoutPlugin)
  localVue.use(ModalPlugin)

  it('should be render', () => {
    const cmp = mount(PollCommandTextArea, {localVue})
    expect(cmp.element).toMatchSnapshot()
  })

  it('the instruction title has the correct value', () => {
    const cmp = mount(PollCommandTextArea, {localVue})

    let instructionTitle = cmp.find('#instructions').text()
    expect(instructionTitle).toEqual('Copy-pasta your command into slack :^)')
  })

  it('the command input has the correct value', () => {
    const cmp = mount(PollCommandTextArea, {localVue, propsData: {
      value: 'I"m command'
    }})

    let commandInput = cmp.find('#command-input').element.value
    expect(commandInput).toEqual('I"m command')
  })

  it('set data in the command input and fire the input event', () => {
    const cmp = mount(PollCommandTextArea, {localVue})

    const commandInput = cmp.find('#command-input')
    commandInput.setValue('I"m command two')

    expect(cmp.emitted().input).toBeTruthy()
    expect(cmp.emitted().input.length).toBe(1)
    expect(cmp.emitted().input[0]).toEqual(['I"m command two'])
  })
})