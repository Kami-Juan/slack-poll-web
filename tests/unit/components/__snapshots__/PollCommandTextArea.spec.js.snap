// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test PollCommandTextArea.vue should be render 1`] = `
<div
  id="poll-command-text-area"
>
  <div
    class="row justify-content-center"
  >
    <div
      class="col-md-6 col-12"
    >
      <span
        id="instructions"
      >
        Copy-pasta your command into slack :^)
      </span>
       
      <textarea
        id="command-input"
      />
    </div>
  </div>
</div>
`;
